### ISSUE TITLE
option-if-let-else  botches map_or_else fn call

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#15002]
Using the following flags

`--force-warn clippy::option-if-let-else`

this code:

```rust
pub fn main() {
    let _: String = match String::from("_").try_into() {
        //~^ useless_conversion
        Ok(a) => a,
        Err(_) => String::new(),
    };
}

```

caused the following diagnostics:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.LzvIlMhAoqCp/icemaker_clippyfix_tempdir.FkCeJpXuwaPo/_a)
warning: use Option::map_or_else instead of an if let/else
 --> src/main.rs:2:21
  |
2 |       let _: String = match String::from("_").try_into() {
  |  _____________________^
3 | |         //~^ useless_conversion
4 | |         Ok(a) => a,
5 | |         Err(_) => String::new(),
6 | |     };
  | |_____^ help: try: `String::from("_").try_into().map_or_else(String::new, |a| a)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#option_if_let_else
  = note: requested on the command line with `--force-warn clippy::option-if-let-else`

warning: `_a` (bin "_a") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.13s

```

However after applying these diagnostics, the resulting code:

```rust
pub fn main() {
    let _: String = String::from("_").try_into().map_or_else(String::new, |a| a);
}

```
no longer compiled:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.LzvIlMhAoqCp/icemaker_clippyfix_tempdir.FkCeJpXuwaPo/_a)
error[E0593]: function is expected to take 1 argument, but it takes 0 arguments
   --> src/main.rs:2:62
    |
2   |     let _: String = String::from("_").try_into().map_or_else(String::new, |a| a);
    |                                                  ----------- ^^^^^^^^^^^ expected function that takes 1 argument
    |                                                  |
    |                                                  required by a bound introduced by this call
    |
note: required by a bound in `std::result::Result::<T, E>::map_or_else`
   --> /home/matthias/.rustup/toolchains/master/lib/rustlib/src/rust/library/core/src/result.rs:854:30
    |
854 |     pub fn map_or_else<U, D: FnOnce(E) -> U, F: FnOnce(T) -> U>(self, default: D, f: F) -> U {
    |                              ^^^^^^^^^^^^^^ required by this bound in `Result::<T, E>::map_or_else`

For more information about this error, try `rustc --explain E0593`.
error: could not compile `_a` (bin "_a") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_a` (bin "_a" test) due to 1 previous error

```

Version:
```
rustc 1.89.0-nightly (9f0e5d963 2025-06-06)
binary: rustc
commit-hash: 9f0e5d963d05ebcf7ae1ca1f4bda3668c702acda
commit-date: 2025-06-06
host: x86_64-unknown-linux-gnu
release: 1.89.0-nightly
LLVM version: 20.1.5

```
